src/react/ui-elements/Veeam/VeeamTable.test.tsx

```diff
@@ -1,40 +1,189 @@
+  }),
+  rest.put('*', (_, res, ctx) => res(ctx.status(500))),
+  rest.patch('*', (_, res, ctx) => res(ctx.status(500))),
+  rest.delete('*', (_, res, ctx) => res(ctx.status(500))),
+  rest.head('*', (_, res, ctx) => res(ctx.status(500))),
+];
+const goodHandlers = [
+  ...getVeeamMutationHandler(),
+  getConfigOverlay(TEST_API_BASE_URL, INSTANCE_ID),
+];
+jest.setTimeout(500_000);
 describe('VeeamTable', () => {
+  const server = setupServer();
+  beforeAll(() => {
+    server.listen({ onUnhandledRequest: 'error' });
+    mockOffsetSize(600, 800);
+  });
+  afterEach(() => {
+    // reset the event listen between the tests
+    server.events.removeAllListeners();
+    server.resetHandlers();
+  });
+  afterAll(() => server.close());
+
   const selectors = {
-    cancelButton: () => screen.getByRole('button', { name: /Cancel/i }),
+    setBucketName: () => screen.getByLabelText(/Bucket name*/i),
+    cancelButton: () => screen.getByRole('button', { name: /Exit/i }),
     continueButton: () => screen.getByRole('button', { name: /Continue/i }),
+    veeamConfigActionTable: () =>
+      screen.getByText('Configure ARTESCA for Veeam'),
+    allRows: () => screen.getAllByRole('row').filter((_, index) => index > 0),
+    retryButton: () => screen.getByRole('button', { name: /retry/i }),
   };
 
-  it('should render the Veeam table', async () => {
+  const setupTest = () => {
     render(
-      <QueryClientProvider client={new QueryClient()}>
-        <Stepper
-          steps={[
-            {
-              label: 'Repository Table',
-              Component: VeeamTable,
-            },
-          ]}
-        />
-      </QueryClientProvider>,
+      <Stepper
+        steps={[
+          {
+            label: 'Configuration',
+            Component: Configuration,
+          },
+          {
+            label: 'Action Status',
+            Component: VeeamTable,
+          },
+        ]}
+      />,
+      { wrapper: NewWrapper() },
     );
+  };
+
+  it('should retry the failed actions', async () => {
+    //Setup
+    server.use(...allFailHandlers);
+    setupTest();
+    //Exercise
+    userEvent.type(selectors.setBucketName(), bucketName);
+    await waitFor(() => {
+      expect(selectors.continueButton()).toBeEnabled();
+    });
+    userEvent.click(selectors.continueButton());
+
+    // Veeam action table
+    await waitFor(() => {
+      expect(selectors.veeamConfigActionTable()).toBeInTheDocument();
+    });
 
+    //Verify
+    // initial state
+    selectors.allRows().forEach((row, index) => {
+      const cells = within(row).getAllByRole('gridcell');
+      expect(cells).toHaveLength(3);
+      expect(cells[0]).toHaveTextContent(`${index + 1}`);
+      expect(cells[1]).toHaveTextContent(actions[index]);
+
+      if (index === 0) {
+        try {
+          expect(cells[2]).toHaveTextContent(/Pending.../i);
+        } catch (error) {
+          expect(cells[2]).toHaveTextContent(/Failed/i);
+        }
+      }
+    });
```

Comment:
nit: maye you want to extarct this to some function "expectInitialState" and mutualize it with the other test ?